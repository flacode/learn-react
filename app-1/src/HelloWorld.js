import React, { Component }  from 'react';
import './HelloWorld.css';

class HelloWorld extends Component{

    constructor(props){
        super(props); // constructor for parent
        this.state = {greeting: 'Hello'};
        this.frenchify = this.frenchify.bind(this);// bind the event to an instance This tells Javascript “Hey, any time you see a reference to this inside of the frenchify function, I want you specifically to refer to ME.” 
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    frenchify(){
        this.setState({greeting: 'Bonjour'});
    }

    removeGreeting(){
        this.props.removeGreeting(this.props.name);
    }

    render(){
        return (
        <div className="HelloWorld">
            {this.state.greeting} { this.props.name }
            <br/>
            <button className="frenchify" onClick={ this.frenchify }>Frenchify!</button>
            <button className="remove" onClick={this.removeGreeting}>Remove Me!</button>
            </div>
        );
    }
}

export default HelloWorld;