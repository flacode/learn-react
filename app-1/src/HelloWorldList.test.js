import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'; // we start off by calling renderer.create, and passing to that function the JSX that we want to build into our new snapshot. 

import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';
import HelloWorld from './HelloWorld';

configure({adapter: new Adapter() });

describe(HelloWorldList, () => {
    const component = shallow(
      <HelloWorldList />
    );
  
    it('renders and matches our snapshot', () => {
      const component = renderer.create(
        <HelloWorldList />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('contains an AddGreeter subcomponent', () => {
      expect(component.find(AddGreeter)).toHaveLength(1);
    });
  
    it('contains the same number of HelloWorld components as greetings', () => {
      const helloWorlds = component.find(HelloWorld).length;
      const greetings = component.state('greetings').length;
      expect(helloWorlds).toEqual(greetings);
    });
  
    it('adds another greeting when the add greeting function is called', () => {
      const before = component.find(HelloWorld).length;
      component.instance().addGreeting('Sample');
      const after = component.find(HelloWorld).length;
      expect(after).toBeGreaterThan(before);
    });
  
    it('removes a greeting from the list when the remove greeting function is called', () => {
      const before = component.find(HelloWorld).length;
      const removeMe = component.state('greetings')[0];
      component.instance().removeGreeting(removeMe);
      const after = component.find(HelloWorld).length;
      expect(after).toBeLessThan(before);
    });
  });