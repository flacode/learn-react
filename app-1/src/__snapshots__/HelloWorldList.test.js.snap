exports[`function HelloWorldList(props) {/* istanbul ignore next */_classCallCheck(this, HelloWorldList);var _this = _possibleConstructorReturn(this, (HelloWorldList.__proto__ || Object.getPrototypeOf(HelloWorldList)).call(this,
        props));
        /* istanbul ignore next */_this.state = { greetings: ["Flavia", "Mary", "Nshemerirwe"] };
        /* istanbul ignore next */_this.addGreeting = /* istanbul ignore next */_this.addGreeting.bind( /* istanbul ignore next */_this);
        /* istanbul ignore next */_this.removeGreeting = /* istanbul ignore next */_this.removeGreeting.bind( /* istanbul ignore next */_this); /* istanbul ignore next */return _this;
    } renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList">
  <div
    className="AddGreeter">
    <input
      onChange={[Function]}
      type="text"
      value="" />
      
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Flavia
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Mary
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Nshemerirwe
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
</div>
`;
